simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderActivity.java

```diff
@@ -588,6 +612,50 @@ else if (id == 1) {
     hs.startIfNecessaryForPackage(p, this);
   }
 
+  private void addBookmark(final @NonNull ReaderBookLocation bookmark) {
+
+    //TODO find ways that an invalid bookmark location could exist and inform user it cannot be made/saved
+
+    final String annot_context = "http://www.w3.org/ns/anno.jsonld";
+    final String type = "Annotation";
+    final String motivation = "http://www.w3.org/ns/oa#bookmarking";
+    final String bookID = this.book_id.toString();
+    final String selectorType = "oa:FragmentSelector";
+    final String value = bookmark.toJsonString();
+    final String timestamp = new Instant().toString();
+    final OptionType<String> deviceID = this.credentials.getAdobeDeviceID().map(
+        id -> id.toString()
+    );
+
+    //TODO GSON-generated classes are not aware of nullability, and are not correctly representing the "body" property
```

Comment:
I don't believe BookAnnotation was generated correctly, because the "body" node is, in its current implementation, supposed to just be set after the constructor as a JsonElement.